src/builder.rs

```diff
@@ -178,6 +175,284 @@ impl<'ctx> Builder<'ctx> {
+    /// let fn_type = f32_type.fn_type(&[], false);
+    ///
+    /// // we will pretend this function can throw an exception
+    /// let function = module.add_function("bomb", fn_type, None);
+    /// let basic_block = context.append_basic_block(function, "entry");
+    ///
+    /// builder.position_at_end(basic_block);
+    ///
+    /// let pi = f32_type.const_float(::std::f64::consts::PI);
+    ///
+    /// builder.build_return(Some(&pi));
+    ///
+    /// let function2 = module.add_function("wrapper", fn_type, None);
+    /// let basic_block2 = context.append_basic_block(function2, "entry");
+    ///
+    /// builder.position_at_end(basic_block2);
+    ///
+    /// let then_block = context.append_basic_block(function2, "then_block");
+    /// let catch_block = context.append_basic_block(function2, "catch_block");
+    ///
+    /// let call_site = builder.build_invoke(function, &[], then_block, catch_block, "get_pi");
+    ///
+    /// {
+    ///     builder.position_at_end(then_block);
+    ///
+    ///     // in the then_block, the `call_site` value is defined and can be used
+    ///     let result = call_site.try_as_basic_value().left().unwrap();
+    ///
+    ///     builder.build_return(Some(&result));
+    /// }
+    ///
+    /// {
+    ///     builder.position_at_end(catch_block);
+    ///
+    ///     // the personality function used by C++
+    ///     let personality_function = {
+    ///         let name = "__gxx_personality_v0";
+    ///
+    ///         module.add_function(name, context.i64_type().fn_type(&[], false), None)
+    ///     };
+    ///
+    ///     // type of an exception in C++
+    ///     let i8_ptr_type = context.i32_type().ptr_type(AddressSpace::Generic);
+    ///     let i32_type = context.i32_type();
+    ///     let exception_type = context.struct_type(&[i8_ptr_type.into(), i32_type.into()], false);
+    ///
+    ///     let res = builder.build_catch_all_landing_pad(exception_type, personality_function, "res");
+    ///
+    ///     // we handle the exception by returning a default value
+    ///
+    ///     builder.build_return(Some(f32_type.const_zero));
+    /// }
+    /// ```
+    pub fn build_catch_all_landing_pad<T>(
+        &self,
+        exception_type: T,
+        personality_function: FunctionValue<'ctx>,
+        name: &str,
+    ) -> BasicValueEnum<'ctx>
+    where
+        T: BasicType<'ctx>,
+    {
+        let c_string = to_c_str(name);
+        let num_clauses = 1u32;
+
+        let value = unsafe {
+            LLVMBuildLandingPad(
+                self.builder,
+                exception_type.as_type_ref(),
+                personality_function.as_value_ref(),
+                num_clauses,
+                c_string.as_ptr(),
+            )
+        };
+
+        // we will add one clause, a null pointer of the i8 type
+        // this is how c++ encodes the `catch(...)` case.
+        let i8_type = unsafe { crate::types::IntType::new(llvm_sys::core::LLVMInt8Type()) };
```

Comment:
is this OK?

Reply:
At minimum, this is incorrect because `LLVMInt8Type` assumes the global context

Reply:
Isn't it possible to do something like:

```rust
exeception_type.get_context().i8_type();
```

I know the `get_context` methods have not so great lifetimes associated with them (bug), but I think that won't be a problem here because we don't save it anywhere

Reply:
well it works in the test... Anyway in this case you need to construct a null pointer of any type really. Is there a way to do that?

Reply:
Also, the comment doesn't make it clear why this needs to be done. Something about exception catching was all I got out of it. But why does it need one clause?

Reply:
or `personality_function.get_context()`

Reply:
yes that worked!